telegram/request/_httpxrequest.py

```diff
@@ -107,6 +102,20 @@ class HTTPXRequest(BaseRequest):
                 these concepts.
 
             .. versionadded:: NEXT.VERSION
+        proxy (:obj:`str` | ``httpx.Proxy`` | ``httpx.URL``, optional): The URL to a proxy server,
+            a ``httpx.Proxy`` object or a ``httpx.URL`` object. For example
+            ``'http://127.0.0.1:3128'`` or ``'socks5://127.0.0.1:3128'``. Defaults to :obj:`None`.
+
+            Note:
+                * The proxy URL can also be set via the environment variables ``HTTPS_PROXY`` or
+                  ``ALL_PROXY``. See `the docs of httpx`_ for more info.
+                * For Socks5 support, additional dependencies are required. Make sure to install
+                  PTB via :command:`pip install "python-telegram-bot[socks]"` in this case.
+                * Socks5 proxies can not be set via environment variables.
```

Comment:
I would also add a note here linking to #3844 as an example of advanced use cases, since the httpx documentation is non existant for `httpx.Proxy` and `httpx.URL`.

Reply:
I added a new bullet point with a description. IMO the discussion in #3844 is a tad to long to be a proper reference in the documentation :D